app/src/main/java/app/revanced/integrations/patches/components/ShortsFilter.java

```diff
@@ -58,22 +59,36 @@ public ShortsFilter() {
                 "shorts_video_cell",
                 "shorts_pivot_item"
         );
+        // Use a different filter group for this pattern, as it requires an additional check after matching.
+        shortsShelf = new StringFilterGroup(
+                SettingsEnum.HIDE_SHORTS,
+                "shelf_header"
+        );
 
         pathFilterGroups.addAll(joinButton, subscribeButton, channelBar, soundButton, infoPanel);
-        identifierFilterGroups.addAll(shorts, thanksButton);
+        identifierFilterGroups.addAll(shorts, shortsShelf, thanksButton);
     }
 
     @Override
     boolean isFiltered(String path, @Nullable String identifier, byte[] protobufBufferArray,
                        FilterGroupList matchedList, FilterGroup matchedGroup, int matchedIndex) {
         if (matchedGroup == soundButton || matchedGroup == infoPanel || matchedGroup == channelBar) return true;
 
-        // Filter the path only when reelChannelBar is visible.
-        if (pathFilterGroups == matchedList) {
-            return path.contains(REEL_CHANNEL_BAR_PATH);
+        if (matchedList == pathFilterGroups) {
+            // Filter the path only when reelChannelBar is visible.
+            if  (!path.contains(REEL_CHANNEL_BAR_PATH)) {
+                return false;
+            }
+        } else if (matchedGroup == shortsShelf) {
+            // Shelf header is used for watch history and possibly other places.
+            // Only hide if the shelf is used for Shorts.
+            if (!path.startsWith("horizontal_video_shelf.eml|")) {
```

Comment:
It should be `path.contains("horizontal_shelf")`, because `library_recent_shelf` was also filtered in logs. Can't test at the moment!

Reply:
Still the same!